dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java

```diff
@@ -408,45 +408,51 @@ public String generateInputJson(String entry, final boolean json) throws ApiExce
+            if (json) {
+                try {
+                    return gson.toJson(stringObjectMap);
+                } catch (CWL.GsonBuildException ex) {
+                    exceptionMessage(ex, "There was an error creating the CWL GSON instance.", API_ERROR);
+                } catch (JsonParseException ex) {
+                    exceptionMessage(ex, "The JSON file provided is invalid.", API_ERROR);
+                }
+            } else {
+                // re-arrange as rows and columns
+                final Map<String, String> typeMap = abstractEntryClient.getCwlUtil().extractCWLTypes(output.getLeft());
+                final List<String> headers = new ArrayList<>();
+                final List<String> types = new ArrayList<>();
+                final List<String> entries = new ArrayList<>();
+                for (final Map.Entry<String, Object> objectEntry : stringObjectMap.entrySet()) {
+                    headers.add(objectEntry.getKey());
+                    types.add(typeMap.get(objectEntry.getKey()));
+                    Object value = objectEntry.getValue();
+                    if (value instanceof Map) {
+                        Map map = (Map)value;
+                        if (map.containsKey("class") && "File".equals(map.get("class"))) {
+                            value = map.get("path");
+                        }
 
+                    }
+                    entries.add(value.toString());
                 }
-                entries.add(value.toString());
-            }
-            final StringBuffer buffer = new StringBuffer();
-            try (CSVPrinter printer = new CSVPrinter(buffer, CSVFormat.DEFAULT)) {
-                printer.printRecord(headers);
-                printer.printComment("do not edit the following row, describes CWL types");
-                printer.printRecord(types);
-                printer.printComment("duplicate the following row and fill in the values for each run you wish to set parameters for");
-                printer.printRecord(entries);
+                final StringBuffer buffer = new StringBuffer();
+                try (CSVPrinter printer = new CSVPrinter(buffer, CSVFormat.DEFAULT)) {
+                    printer.printRecord(headers);
+                    printer.printComment("do not edit the following row, describes CWL types");
+                    printer.printRecord(types);
+                    printer.printComment("duplicate the following row and fill in the values for each run you wish to set parameters for");
+                    printer.printRecord(entries);
+                }
+                return buffer.toString();
             }
-            return buffer.toString();
+            return null;
+        } catch (ClassCastException ex) {
+            errorMessage("Cannot generate a JSON for this workflow.\n"
+                    + "The structure of its input is currently not supported. Please provide items as a String.", GENERIC_ERROR);
```

Comment:
Is the workaround to have:

```
type: string[]
```

instead of 

```
type: array
items: string
```

If yes, your message could be a little more specific. Also, I think (?) CWL is case-sensitive; if so, you should have `string` (`string[]`?) instead of `String`.

Reply:
Due to the [casting](https://github.com/dockstore/cwlavro/blob/83e0bd298060472bea42d638460c5899a67380d0/cwlavro-tools/src/main/java/io/cwl/avro/CWL.java#L307-L308) here, both `type` and `items` should be strings. I edited the message to say `string` instead of `String`